segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/ThrottlerCalculator.java

```diff
@@ -41,22 +42,26 @@
     @VisibleForTesting
     static final int MAX_DELAY_MILLIS = 25000;
     /**
-     * Amount of time (millis) to increase throttling by for each percentage point increase in the cache utilization (above 100%).
+     * Adjustment to the target utilization ratio at or above which throttling will begin to apply. This value helps
```

Comment:
Isn't the description and example descriptive enough? What is says is this:
1. We will do our best not to throttle if we can maintain performance by other means (other means == evict non-pinned items from the cache).
2. The cache manager will evict items at the specified target utilization (in the config file) and try to keep the cache size below that limit.
3. If the cache manager fails to do that, our last resort is to slow down the ingestion rate so that we can give downstream components a chance to unpin cache items (transfer data to tier2).

If you set target=max, then you are essentially putting a high wall with no steps to climb on it. You will hit that wall and your ingestion performance will immediately drop to zero. The goal of this adjustment is to give the cache manager enough breathing room so that it can do its job. It's an async task so the cache utilization may slightly exceed the target until it can reduce it below. To avoid unnecessary wobbling in throttling, I realized there's a need for a buffer between this target and when we begin throttling.

Reply:
The example makes clear what is happening, but the why is missing. The argument about the separation of goals does not justify well the need for an adjustment. The insight you gave me in your answer is much more illustrative.